contracts/tests/test_peer_protocol.cairo

```diff
@@ -154,3 +157,75 @@ fn test_withdraw() {
 
     stop_cheat_caller_address(peer_protocol_address);
 }
+
+#[test]
+fn test_get_user_deposits() {
+    // Deploy contracts
+    let token1_address = deploy_token("MockToken1");
+    let token2_address = deploy_token("MockToken2");
+    // add another supported token without balance
+    let token3_address = deploy_token("MockToken3");
+    let peer_protocol_address = deploy_peer_protocol();
+
+    // Setup dispatchers
+    let token1 = IERC20Dispatcher { contract_address: token1_address };
+    let token2 = IERC20Dispatcher { contract_address: token2_address };
+    let peer_protocol = IPeerProtocolDispatcher { contract_address: peer_protocol_address };
+
+    // Setup addresses
+    let owner: ContractAddress = starknet::contract_address_const::<0x123626789>();
+    let user: ContractAddress = starknet::contract_address_const::<0x122226789>();
+
+    // Define amounts
+    let mint_amount: u256 = 1000 * ONE_E18;
+    let deposit_amount1: u256 = 100 * ONE_E18;
+    let deposit_amount2: u256 = 200 * ONE_E18;
+
+    // Add supported tokens
+    start_cheat_caller_address(peer_protocol_address, owner);
+    peer_protocol.add_supported_token(token1_address);
+    peer_protocol.add_supported_token(token2_address);
+    peer_protocol.add_supported_token(token3_address);
+    stop_cheat_caller_address(peer_protocol_address);
+
+    // Mint tokens to user
+    token1.mint(user, mint_amount);
+    token2.mint(user, mint_amount);
+
+    // Approve spending
+    start_cheat_caller_address(token1_address, user);
+    token1.approve(peer_protocol_address, mint_amount);
+    stop_cheat_caller_address(token1_address);
+
+    start_cheat_caller_address(token2_address, user);
+    token2.approve(peer_protocol_address, mint_amount);
+    stop_cheat_caller_address(token2_address);
+
+    // Make deposits
+    start_cheat_caller_address(peer_protocol_address, user);
+    peer_protocol.deposit(token1_address, deposit_amount1);
+    peer_protocol.deposit(token2_address, deposit_amount2);
+
+    // Get and verify user deposits
+    let user_deposits = peer_protocol.get_user_deposits(user);
+
+    assert!(user_deposits.len() == 2, "incorrect number of deposits");
+
+    // Verify deposit amounts for each token
+    let deposit1 = user_deposits.at(0);
+    let deposit2 = user_deposits.at(1);
+
+    assert!(
+        (*deposit1.token == token1_address && *deposit1.amount == deposit_amount1)
+            || (*deposit1.token == token2_address && *deposit1.amount == deposit_amount2),
+        "incorrect deposit1"
+    );
+    assert!(
+        (*deposit2.token == token1_address && *deposit2.amount == deposit_amount1)
+            || (*deposit2.token == token2_address && *deposit2.amount == deposit_amount2),
+        "incorrect deposit2"
+    );
+    assert!(deposit1.token != deposit2.token, "duplicate tokens in result");
+
+    stop_cheat_caller_address(peer_protocol_address);
+}
```

Comment:
Don't you think that testing for zero address in a getter method is actually unnecessary overhead because all deposit operations require valid token transfers, and ERC20 and deposit method already validate addresses during deposits?